src/lib/frontend/d_cnf.ml

```diff
@@ -416,24 +439,60 @@ let fpa_builtins =
     |> op "not_theory_constant" Not_theory_constant ([real] ->. prop)
     |> is_theory_constant
     |> op "linear_dependency" Linear_dependency ([real; real] ->. prop)
+  in
+  fun env s ->
+    match s with
+    | Dl.Typer.T.Id id ->
+      begin
+        try
+          Id.Map.find_exn id fpa_builtins env s
+        with Not_found -> `Not_found
+      end
+    | Builtin _ -> `Not_found
 
+let smt_fpa_builtins =
+  let term_app1 env s f =
+    Dl.Typer.T.builtin_term @@
+    Dolmen_type.Base.term_app1 (module Dl.Typer.T) env s f
+  in
+  let other_builtins =
+    Id.Map.empty
+    |> smt_add_rounding_modes
   in
   fun env s ->
-    begin match s with
-      | Dl.Typer.T.Id id ->
-        begin
-          try
-            Id.Map.find_exn id fpa_builtins env s
-          with Not_found -> `Not_found
-        end
-      | Builtin _ -> `Not_found
-    end
+    match s with
+    | Dl.Typer.T.Id {
+        ns = Term ;
+        name = Indexed {
+            basename = "ae.round" ;
+            indexes = [ i; j; rm ] } } ->
```

Comment:
I was focused on the cache earlier, but: `ae.round` should only have two indexed identifiers; the rounding mode argument should be an actual term-level argument (in the same way that in the FloatingPoints SMT-LIB theory only the bit widths are indexed).